File: examples/ocaml/x05.ml
(*
        Histogram demo.
*)

open Plplot

let pi = atan 1.0 *. 4.0

let npts = 2047

(*--------------------------------------------------------------------------*\
 * Draws a histogram from sample data.
\*--------------------------------------------------------------------------*)

let () =
  (* Parse and process command line arguments *)
  plparseopts Sys.argv [PL_PARSE_FULL];

  (* Initialize plplot *)
  plinit ();

  (* Fill up data points *)

  let delta = 2.0 *. pi /. float_of_int npts in
  let data = Array.init npts (fun i -> sin (float_of_int i *. delta)) in

  plcol0 1;
  plhist data (-1.1) 1.1 44 [PL_HIST_DEFAULT];
  plcol0 2;
  pllab "#frValue" "#frFrequency"
    "#frPLplot Example 5 - Probability function of Oscillator";

  plend ();
  ()

File: examples/python/x05.py
#! /usr/local/bin/python
#
#	x05c.c
#
#	Histogram demo.

import math
import pl
import sys

NPTS = 2047

# main
#
# Draws a histogram from sample data.

def main():

	# Parse and process command line arguments

	pl.ParseOpts(sys.argv, pl.PARSE_FULL)

	# Initialize plplot

	pl.init()

	# Fill up data points

	delta = 2.0 * math.pi / NPTS
	data = []
	for i in range(NPTS):
		data.append(math.sin(i * delta))

	pl.col(1)
	pl.hist(data, -1.1, 1.1, 44, 0)
	pl.col(2)
	pl.lab("#frValue", "#frFrequency",
	       "#frPLplot Example 5 - Probability function of Oscillator")

	pl.end()

main()
